modules/base/src/processors/volumevoronoisegmentation.cpp

```diff
@@ -0,0 +1,104 @@
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/base/processors/volumevoronoisegmentation.h>
+#include <modules/base/algorithm/volume/volumevoronoi.h>
+
+namespace inviwo {
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo VolumeVoronoiSegmentation::processorInfo_{
+    "org.inviwo.VolumeVoronoiSegmentation",  // Class identifier
+    "Volume Voronoi Segmentation",           // Display name
+    "Undefined",                             // Category
+    CodeState::Experimental,                 // Code state
+    Tags::None,                              // Tags
+};
+const ProcessorInfo VolumeVoronoiSegmentation::getProcessorInfo() const { return processorInfo_; }
+
+VolumeVoronoiSegmentation::VolumeVoronoiSegmentation()
+    : Processor(), volume_("inputVolume"), inputMesh_("inputPositions"), outport_("outport") {
+
+    addPort(volume_);
+    addPort(inputMesh_);
+    addPort(outport_);
+}
+
+void VolumeVoronoiSegmentation::process() {
+    auto inputMesh = inputMesh_.getData();
+
+    // Get seed points
+    // TODO: see if this can be done in a better way
+    auto posIt = util::find_if(inputMesh->getBuffers(), [](const auto& buf) {
+        return buf.first.type == BufferType::PositionAttrib;
+    });
+    if (posIt == inputMesh->getBuffers().end()) {
+        return;
+    }
+    const auto positionRam = posIt->second->getRepresentation<BufferRAM>();
+    std::vector<dvec3> positions;
+    for (std::size_t i = 0; i < positionRam->getSize(); ++i) {
+        positions.push_back(positionRam->getAsDVec3(i));
+    }
+
+    // Get indices for seed points (if exists??)
+    // TODO: should probably have some fallback solution if there are no indices...?
+    auto indexIt = util::find_if(inputMesh->getBuffers(), [](const auto& buf) {
+        return buf.first.type == BufferType::IndexAttrib;
+    });
+    if (indexIt == inputMesh->getBuffers().end()) {
+        return;
+    }
+    // TODO: see if this can be done in a better way
+    const auto& indices = static_cast<const BufferRAMPrecision<uint32_t, BufferTarget::Data>*>(
+                              indexIt->second->getRepresentation<BufferRAM>())
+                              ->getDataContainer();
+
+    if (positions.size() != indices.size()) {
+        return;
```

Comment:
generally when something goes wrong we would throw and exception with a message
`throw Exception("Unexpected dimension missmatch", IVW_CONTEXT);`
the last argument `IVW_CONTEXT` is a macro that will add the filename, line and function to the error message automatically.

Reply:
Ok thanks! I am now using that pattern instead